src/azure-cli/azure/cli/command_modules/util/custom.py

```diff
@@ -175,6 +175,51 @@ def upgrade_version(cmd, update_all=None, yes=None):  # pylint: disable=too-many
                    else auto_upgrade_msg)
 
 
+def _upgrade_on_windows():
+    """Download MSI to a temp folder and install it with msiexec.exe.
+    Directly installing from URL may be blocked by policy: https://github.com/Azure/azure-cli/issues/19171
+    This also gives the user a chance to manually install the MSI in case of msiexec.exe failure.
+    """
+    logger.warning("Updating Azure CLI with MSI from https://aka.ms/installazurecliwindows")
```

Comment:
This was `logger.debug` before. Intentional change?

Reply:
Yes.

I want the command to be more explicit and give the user more information about what it is doing internally. Otherwise, in case of failure, all that the user can see is `CLI upgrade failed`. (No different from `Something occurred`.)

To view the details, the user must specify `--debug` which is definitely not user-friendly.